pytorch_lightning/utilities/data.py

```diff
@@ -58,18 +56,29 @@ def extract_batch_size(batch: BType) -> int:
     Returns:
         ``len(tensor)`` when found, or ``1`` when it hits an empty or non iterable.
     """
-    batch_size = None
-    for bs in _extract_batch_size(batch):
+    error_msg = (
+        "Trying to infer `batch_size` from an ambiguous collection but couldn't find one. Either"
+        " set `self.log(..., batch_size=batch_size)` or simplify the batch collection passed"
+        " inside `*_step` functions."
```

Comment:
When would the user "simplify the batch collection passed..."?

If the user has that collection structure, it's probably for a reason

Reply:
sometimes they are not aware of how the collate_fn works so I guess it's good to have.